compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -328,10 +329,26 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             let lltrue = helper.llbb_with_cleanup(self, target);
             let llfalse = helper.llbb_with_cleanup(self, targets.otherwise());
             if switch_ty == bx.tcx().types.bool {
+                let expect = if let Some(x) = self.mir[bb].statements.last()
+                    && let mir::StatementKind::Intrinsic(box mir::NonDivergingIntrinsic::Expect(
+                        op,
+                        kind,
+                    )) = &x.kind
+                    && self.codegen_operand(bx, op).immediate() == discr.immediate()
+                {
+                    match kind {
+                        mir::ExpectKind::True => ExpectKind::True,
+                        mir::ExpectKind::False => ExpectKind::False,
+                        mir::ExpectKind::Unpredictable => ExpectKind::Unpredictable,
+                    }
+                } else {
+                    ExpectKind::None
+                };
```

Comment:
Nit: could this be a `Option<mir::ExpectKind>`?

Reply:
Good point. Updated